thinkhazard/models.py

```diff
@@ -18,10 +17,50 @@
 Base = declarative_base()
 
 
-class MyModel(Base):
-    __tablename__ = 'models'
+class Admin(AbstractConcreteBase, Base):
+    __table_args__ = {'schema': 'adminunits'}
+
     id = Column(Integer, primary_key=True)
-    name = Column(Text)
-    value = Column(Integer)
+    name = Column(Unicode)
+
+    def __init__(self, name):
+        self.name = name
+
+
+class Admin0(Admin):
+    __tablename__ = 'admin0'
+    order = 0
+
+    def __json__(self, request):
+        return {'admin0': self.name}
+
+
+class Admin1(Admin):
+    __tablename__ = 'admin1'
+    order = 1
+
+    admin0_name = Column(Unicode)
+
+    def __init__(self, name, admin0_name):
+        super(Admin1, self).__init__(name)
+        self.admin0_name = admin0_name
+
+    def __json__(self, request):
+        return {'admin0': self.admin0_name, 'admin1': self.name}
+
+
+class Admin2(Admin):
+    __tablename__ = 'admin2'
+    order = 2
+
+    admin0_name = Column(Unicode)
```

Comment:
I agree this denormalization (admin0 field on admin2) is a good idea for performance reasons. Is there a need for a constraint or a trigger at the database level to ensure it stays in sync?